src/frontend/src/modals/IOModal/components/chatView/chatInput/newChatInput.tsx

```diff
@@ -0,0 +1,214 @@
+        ...prevFiles,
+        { file, loading: true, error: false, id, type },
+      ]);
+
+      mutate(
+        { file, id: currentFlowId },
+        {
+          onSuccess: (data) => {
+            setFiles((prev) => {
+              const newFiles = [...prev];
+              const updatedIndex = newFiles.findIndex((file) => file.id === id);
+              newFiles[updatedIndex].loading = false;
+              newFiles[updatedIndex].path = data.file_path;
+              return newFiles;
+            });
+          },
+          onError: (error) => {
+            setFiles((prev) => {
+              const newFiles = [...prev];
+              const updatedIndex = newFiles.findIndex((file) => file.id === id);
+              newFiles[updatedIndex].loading = false;
+              newFiles[updatedIndex].error = true;
+              return newFiles;
+            });
+            setErrorData({
+              title: "Error uploading file",
+              list: [error.response?.data?.detail],
+            });
+          },
+        },
+      );
+    }
+
+    if ("target" in event && event.target instanceof HTMLInputElement) {
+      event.target.value = "";
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener("paste", handleFileChange);
+    return () => {
+      document.removeEventListener("paste", handleFileChange);
+    };
+  }, [handleFileChange, currentFlowId, lockChat]);
+
+  const send = () => {
+    sendMessage({
+      repeat: 1,
+      files: files.map((file) => file.path ?? "").filter((file) => file !== ""),
+    });
+    setFiles([]);
+  };
+
+  const checkSendingOk = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    return (
+      event.key === "Enter" &&
+      !lockChat &&
+      !event.shiftKey &&
+      !event.nativeEvent.isComposing
+    );
+  };
+
+  const classNameFilePreview = `flex w-full items-center gap-2 bg-background py-2 overflow-auto custom-scroll`;
+
+  const handleButtonClick = () => {
+    fileInputRef.current!.click();
+  };
+
+  return (
+    <div className="flex w-full flex-col-reverse">
+      <div className="flex w-full flex-col rounded-md border border-border p-4">
+        <TextAreaWrapper
+          checkSendingOk={checkSendingOk}
+          send={send}
+          lockChat={lockChat}
+          noInput={noInput}
+          chatValue={chatValue}
+          setChatValue={setChatValue}
+          CHAT_INPUT_PLACEHOLDER={CHAT_INPUT_PLACEHOLDER}
+          CHAT_INPUT_PLACEHOLDER_SEND={CHAT_INPUT_PLACEHOLDER_SEND}
+          inputRef={inputRef}
+          setInputFocus={setInputFocus}
+          files={files}
+          isDragging={isDragging}
+        />
+        <div className={classNameFilePreview}>
+          {files.map((file) => (
+            <FilePreview
+              error={file.error}
+              file={file.file}
+              loading={file.loading}
+              key={file.id}
+              onDelete={() => {
+                setFiles((prev: FilePreviewType[]) =>
```

Comment:
can we have a function to each interaction inside the HTML, to improve readability?